helper/config/decode.go

```diff
@@ -154,3 +156,30 @@ func uint8ToStringHook(f reflect.Kind, t reflect.Kind, v interface{}) (interface
 
 	return v, nil
 }
+
+func stringToTrilean(f reflect.Type, t reflect.Type, v interface{}) (interface{}, error) {
+	// We have a custom data type, trilean, which we read from a string and
+	// then cast to a *bool. Why? So that we can appropriately read "unset"
+	// *bool values in order to intelligently default, even when the values are
+	// being set by a template variable.
+
+	testTril, _ := trilean.FromString("")
+	if t == reflect.TypeOf(testTril) {
+		// From value is string
+		if f == reflect.TypeOf("") {
```

Comment:
No important: 
here you can also use
```suggestion
		switch t {
		  case reflect.String:
```
as `t` is the type of `f`.

Reply:
I don't think this is possible. `f` is a reflect.Type, not a reflect.Kind. Wo your suggestion doesn't work. And since we only have the type itself, I can't use the `type` keyword in the switch statement to switch on the string and bool types themselves. I'd have to use reflect.ValueOf on v and then switch on v.(type) and that seems like just as much work as what I'm doing here.

Reply:
Oh, right I got all confused with `uint8ToStringHook` 👍